packages/core/contracts/data-verification-mechanism/implementation/VotingV2.sol

```diff
@@ -1066,12 +1062,18 @@ contract VotingV2 is Staker, OracleInterface, OracleAncillaryInterface, OracleGo
         returns (RequestStatus)
     {
         if (priceRequest.lastVotingRound == 0) return RequestStatus.NotRequested;
-        else if (priceRequest.lastVotingRound < currentRoundId) {
+        if (priceRequest.lastVotingRound < currentRoundId) {
+            // Check if the request has already been resolved
             VoteInstance storage voteInstance = priceRequest.voteInstances[priceRequest.lastVotingRound];
             (bool isResolved, ) = _getResolvedPrice(voteInstance, priceRequest.lastVotingRound);
-            return isResolved ? RequestStatus.Resolved : RequestStatus.Active;
-        } else if (priceRequest.lastVotingRound == currentRoundId) return RequestStatus.Active;
-        else return RequestStatus.Future; // Means than priceRequest.lastVotingRound > currentRoundId
+            if (isResolved) return RequestStatus.Resolved;
+            if (_shouldDeleteRequest(_getActualRollCount(priceRequest, currentRoundId), priceRequest.isGovernance))
+                return RequestStatus.ToDelete;
```

Comment:
Please also update the docstrings for _getRequestStatus to reflect possible returned state